test/jasmine/tests/sunburst_test.js

```diff
@@ -317,6 +342,88 @@ describe('Test sunburst calc:', function() {
         expect(extract('id')).toEqual(['true', '1', '2', '3', '4', '5', '6', '7', '8']);
         expect(extract('pid')).toEqual(['', 'true', 'true', '2', '2', 'true', 'true', '6', 'true']);
     });
+
+    it('should use *marker.colors*', function() {
+        _calc({
+            marker: { colors: ['pink', '#777', '#f00', '#ff0', '#0f0', '#0ff', '#00f', '#f0f', '#fff'] },
+            labels: ['Eve', 'Cain', 'Seth', 'Enos', 'Noam', 'Abel', 'Awan', 'Enoch', 'Azura'],
+            parents: ['', 'Eve', 'Eve', 'Seth', 'Seth', 'Eve', 'Eve', 'Awan', 'Eve']
+        });
+
+        var cd = gd.calcdata[0];
+        expect(cd.length).toEqual(9);
+        expect(cd[0].color).toEqual('rgba(255, 192, 203, 1)');
+        expect(cd[1].color).toEqual('rgba(119, 119, 119, 1)');
+        expect(cd[2].color).toEqual('rgba(255, 0, 0, 1)');
+        expect(cd[3].color).toEqual('rgba(255, 255, 0, 1)');
+        expect(cd[4].color).toEqual('rgba(0, 255, 0, 1)');
+        expect(cd[5].color).toEqual('rgba(0, 255, 255, 1)');
+        expect(cd[6].color).toEqual('rgba(0, 0, 255, 1)');
+        expect(cd[7].color).toEqual('rgba(255, 0, 255, 1)');
+        expect(cd[8].color).toEqual('rgba(255, 255, 255, 1)');
+    });
+
+    it('should use *marker.colors* numbers with colorscale', function() {
+        _calc({
+            marker: { colors: [1, 2, 3, 4, 5, 6, 7, 8, 9], colorscale: 'Portland' },
```

Comment:
Hmm. What about the cases where `colorscale` isn't set?

By the looks of it, `hasColorscale`

https://github.com/plotly/plotly.js/blob/003f52e4829c082547962bb2e1e0096259ca3d4d/src/components/colorscale/helpers.js#L20-L45

returns `true` when `colorscale` is set regardless of if `marker.color(s)` has numeric items.

Reply:
I think that case is covered:
https://github.com/plotly/plotly.js/blob/1c136f4b1257b4a0558f5bc5699f517324b64300/test/jasmine/tests/treemap_test.js#L276-L291

Reply:
> I think that case is covered:

Not really, this default test essentially just asserts that 

https://github.com/plotly/plotly.js/blob/003f52e4829c082547962bb2e1e0096259ca3d4d/src/lib/coerce.js#L172-L174

returns the `Reds` colorscale array - which is overwritten during the calc step under `autocolorscale: true`:

https://github.com/plotly/plotly.js/blob/003f52e4829c082547962bb2e1e0096259ca3d4d/src/components/colorscale/calc.js#L70-L76

---

With `marker.colors: [0]` it _turns out_ that auto colorscale (after the calc step) is "Reds", but try setting `marker.colors: [-1]`: this makes `marker.colorscale` "Reds" after the defaults, but then "Blues" after the calc step.

So, please a calc test w/o a set colorscale. Thanks!

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4242/commits/ba492cf4ce9396b374abfe6f3fdfb020ad1be12c.